vault/eventbus/bus_test.go

```diff
@@ -180,3 +184,177 @@ func TestBus2Subscriptions(t *testing.T) {
 		t.Error("Timeout waiting for event2")
 	}
 }
+
+// TestBusSubscriptionsCancel verifies that canceled subscriptions are cleaned up.
+func TestBusSubscriptionsCancel(t *testing.T) {
+	subscriptions.Store(0)
+	bus, err := NewEventBus(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx := context.Background()
+	bus.Start()
+
+	// create and cancel a bunch of subscriptions
+	create := 100
+	cancel := 50
```

Comment:
nit: Just to make super clear these don't change later

```suggestion
	const create = 100
	const cancel = 50
```